### PR TITLE
{lib}[GCCcore/13.3.0] hwloc v2.10.0, UCX v1.16.0, UCC v1.3.0

### PR DESCRIPTION
**Pull request by SebastianAchilles**
(created using `eb --new-pr`)
Depends on:
- https://github.com/easybuilders/easybuild-easyconfigs/pull/20580

### CODE REVIEW

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
jscclxc1.int.jsc-clx.fz-juelich.de - Linux Rocky Linux 9.4, x86_64, Intel Xeon Processor (Cascadelake) (cascadelake), Python 3.9.18
See https://gist.github.com/SebastianAchilles/38c58ac501c076b725abc4a788b3e4f7 for a full test report.

**Comment by SebastianAchilles** [comment]
@boegelbot please test @ generoso

**Comment by SebastianAchilles** [comment]
@boegelbot please test @ jsc-zen3

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
aarch64-neoverse-n1-node1.int.aws-rocky88-202310.eessi.io - Linux Rocky Linux 8.9, AArch64, ARM UNKNOWN (neoverse_n1), Python 3.6.8
See https://gist.github.com/boegel/4dcfca69aff429cf7031380b6a2cd9fe for a full test report.

**Comment by boegel** [comment]
@boegelbot please test @ jsc-zen3

File: easybuild/easyconfigs/u/UCX/UCX-1.16.0-GCCcore-13.3.0.eb [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+]
+
+osdependencies = [OS_PKG_IBVERBS_DEV]
+
+dependencies = [
+    ('zlib', '1.3.1'),
+    ('numactl', '2.0.18'),
+]
+
+configure_cmd = "contrib/configure-release"
```
Comment (boegel) [inline easybuild/easyconfigs/u/UCX/UCX-1.16.0-GCCcore-13.3.0.eb:40]: @SebastianAchilles Please change this to use the standard configure script, see #20428

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/u/UCX/UCX-1.16.0-GCCcore-13.3.0.eb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+configopts = '--enable-optimizations --enable-cma --enable-mt --with-verbs '
+configopts += '--without-java --without-go --disable-doxygen-doc '
```
Comment (boegel) [inline easybuild/easyconfigs/u/UCX/UCX-1.16.0-GCCcore-13.3.0.eb:43]:
```suggestion
configopts = '--enable-optimizations --enable-cma --enable-mt --with-verbs '
configopts += '--without-java --without-go --disable-doxygen-doc '
# include the configure options from contrib/configure-release
configopts += '--disable-logging --disable-debug --disable-assertions --disable-params-check '
```

**Review by boegel — CHANGES_REQUESTED** [review]

**Comment by SebastianAchilles** [comment]
@boegelbot please test @ jsc-zen3

**Comment by bedroge** [comment]
Test report by @bedroge
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
interactive1 - Linux Rocky Linux 8.9, x86_64, AMD EPYC-Milan Processor (zen2), Python 3.6.8
See https://gist.github.com/bedroge/194ed16c110da21d8dbd5b8c10084cf5 for a full test report.

**Comment by bedroge** [comment]
Test report by @bedroge
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
bob-Latitude-5300 - Linux Ubuntu 22.04, x86_64, Intel(R) Core(TM) i7-8665U CPU @ 1.90GHz, Python 3.10.12
See https://gist.github.com/bedroge/63c2ef35fd7783e1366628a917275387 for a full test report.

File: easybuild/easyconfigs/u/UCX/UCX-1.16.0-GCCcore-13.3.0.eb [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    'UCX-1.13.1-dynamic_modules.patch',
+]
+checksums = [
+    {'ucx-1.16.0.tar.gz': 'f73770d3b583c91aba5fb07557e655ead0786e057018bfe42f0ebe8716e9d28c'},
+    {'UCX-1.13.1-dynamic_modules.patch': '00874687bd90b795fff61aaa183f6c6bea2210aa1003b28f23d9ebf7066f8782'},
+]
+
+# Uncomment when updating to final release
+# sources = ['%(namelower)s-%(version)s.tar.gz']
+builddependencies = [
```
Comment (boegel) [inline easybuild/easyconfigs/u/UCX/UCX-1.16.0-GCCcore-13.3.0.eb:27]:
```suggestion
```

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/u/UCX/UCX-1.16.0-GCCcore-13.3.0.eb [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+patches = [
+    'UCX-1.13.1-dynamic_modules.patch',
+]
+checksums = [
+    {'ucx-1.16.0.tar.gz': 'f73770d3b583c91aba5fb07557e655ead0786e057018bfe42f0ebe8716e9d28c'},
+    {'UCX-1.13.1-dynamic_modules.patch': '00874687bd90b795fff61aaa183f6c6bea2210aa1003b28f23d9ebf7066f8782'},
+]
+
+# Uncomment when updating to final release
+# sources = ['%(namelower)s-%(version)s.tar.gz']
```
Comment (boegel) [inline easybuild/easyconfigs/u/UCX/UCX-1.16.0-GCCcore-13.3.0.eb:26]:
```suggestion
```

**Review by boegel — CHANGES_REQUESTED** [review]

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
jrc0870.jureca - Linux Rocky Linux 8.9, AArch64, ARM UNKNOWN (neoverse_n1), 2 x NVIDIA NVIDIA A100-PCIE-40GB, 535.154.05, Python 3.6.8
See https://gist.github.com/SebastianAchilles/9cb312dae2370195e9cfba07cb94d371 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
aarch64-neoverse-n1-node1.int.aws-rocky88-202310.eessi.io - Linux Rocky Linux 8.9, AArch64, ARM UNKNOWN (neoverse_n1), Python 3.6.8
See https://gist.github.com/boegel/4b000b72cb071e449ed710105be81649 for a full test report.

**Review by boegel — APPROVED** [review]

**Comment by bedroge** [comment]
Test report by @bedroge
**FAILED**
Build succeeded for 15 out of 21 (3 easyconfigs in total)
starfive - Linux Debian GNU/Linux n/a, RISC-V-64, UNKNOWN, Python 3.10.9
See https://gist.github.com/bedroge/2b4b7c0297c06d94aa7ab4819912c65a for a full test report.

**Comment by boegel** [comment]
@bedroge I won't let that problem with `OpenSSL/3` on RISC-V block this PR, please open a dedicated issue to follow up on that one...

**Comment by boegel** [comment]
Going in, thanks @SebastianAchilles!

**Comment by bedroge** [comment]
Test report by @bedroge
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
starfive - Linux Debian GNU/Linux n/a, RISC-V-64, UNKNOWN, Python 3.10.9
See https://gist.github.com/bedroge/3218f46c1e754c2af4eba35c73df95b9 for a full test report.

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
bear-pg0207u28a.bear.cluster - Linux RHEL 8.6, x86_64, AMD EPYC 9554 64-Core Processor (zen4), Python 3.6.8
See https://gist.github.com/branfosj/34917332362444b37ae9025f65ed0a82 for a full test report.

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
bear-pg0211u03a.bear.cluster - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz (cascadelake), Python 3.6.8
See https://gist.github.com/branfosj/5555da7417889096820d8633c6e8aeda for a full test report.

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
bear-pg0105u03a - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), Python 3.6.8
See https://gist.github.com/branfosj/54705d394f20da8b850077bf2e72745f for a full test report.

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
bear-pg0207u20a - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8480CL (sapphirerapids), Python 3.6.8
See https://gist.github.com/branfosj/675fea117a41b908a2669d2f7fc76732 for a full test report.

**Pull request merged**